### PR TITLE
Documenting service_accounts attribute

### PR DESCRIPTION
**Pull request by Temikus**
+ Removed the old tags attribute todo which was implemented.

CC/ @erjohnso 
#46 can be closed now.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -175,6 +175,7 @@ This provider exposes quite a few provider-specific configuration options:
 * `zone` - The zone name where the instance will be created.
 * `can_ip_forward` - Boolean whether to enable IP Forwarding.
 * `external_ip` - The external IP address to use.
+* `service_accounts` - An array of OAuth2 account scopes for the services the instance will have access to, for example: `['https://www.googleapis.com/auth/devstorage.full_control']`
```
Comment (erjohnso) [inline README.md]: What do you think about having vagrant support the `aliases` that are defined by `gcloud compute`?  That would make vagrant more consistent with the command-line utility. See `gcloud compute instances create --help` and look for `scopes`.

```
          Alias              URI
          bigquery           https://www.googleapis.com/auth/bigquery
          cloud-platform     https://www.googleapis.com/auth/cloud-platform
          compute-ro         https://www.googleapis.com/auth/compute.readonly
          compute-rw         https://www.googleapis.com/auth/compute
          computeaccounts-ro https://www.googleapis.com/auth/computeaccounts.readonly
          computeaccounts-rw https://www.googleapis.com/auth/computeaccounts
          datastore          https://www.googleapis.com/auth/datastore
          logging-write      https://www.googleapis.com/auth/logging.write
          monitoring         https://www.googleapis.com/auth/monitoring
          sql                https://www.googleapis.com/auth/sqlservice
          sql-admin          https://www.googleapis.com/auth/sqlservice.admin
          storage-full       https://www.googleapis.com/auth/devstorage.full_control
          storage-ro         https://www.googleapis.com/auth/devstorage.read_only
          storage-rw         https://www.googleapis.com/auth/devstorage.read_write
          taskqueue          https://www.googleapis.com/auth/taskqueue
          userinfo-email     https://www.googleapis.com/auth/userinfo.email
```

**Comment by Temikus** [comment]
@erjohnso, awesome idea! :+1: Implemented.

Fog-google just maps both attributes to the same option:
```
attribute :service_accounts, :aliases => 'serviceAccounts'
```
so, I just aliased attribute reader/writer:
```
      # The list of access controls for service accounts.
      #
      # @return [Array]
      attr_accessor :service_accounts
      alias :aliases :service_accounts
      alias :aliases= :service_accounts=
```
You can even mix and match now. Setting the following in Vagrantfile:
```
zone.scopes = ['bigquery', 'monitoring', 'https://www.googleapis.com/auth/compute']
```
results in:
```
==> default:  -- Scopes:    ["bigquery", "monitoring", "https://www.googleapis.com/auth/compute"]
```

**Comment by erjohnso** [comment]
Great, but I *think* you're going to need to map those aliases to full URLs though (which is what glcoud does if I'm not mistaken).  At least, that's what I did for libcloud,

https://github.com/apache/libcloud/blob/trunk/libcloud/compute/drivers/gce.py#L971
https://github.com/apache/libcloud/blob/trunk/libcloud/compute/drivers/gce.py#L4677

If I'm mistaken, then I guess I get to delete some code. :)

**Comment by Temikus** [comment]
I was _just_ about to write that and already slightly amended the commit.
The thing is, it works when the latter part of the endpoint matches the alias, like:
`https://www.googleapis.com/auth/compute` and `compute`, however, aliases like `sql` don't, since they are mapped to `https://www.googleapis.com/auth/sqlservice`

Mapping the arbitrary parameters to the api endpoints will be a bit trickier, so I want to ask:
DISCLAIMER: I may not know the details of the initial decision, so please excuse my ignorance/repeated questions on this.
1) Wouldn't maintaining arbitrary aliases make a lot of work for us in the future? Syncing and maintaining a list of arbitrary words across multiple projects can turn into quite a headache. Especially if the tool gets totally rewritten or deprecated.
Additionally, what comes to mind is:
- There is no unified schema to OAuth endpoints, so we cannot autoremap them:
`https://www.googleapis.com/auth/devstorage.full_control` vs `https://www.googleapis.com/auth/sqlservice.admin` for example.
- The end the difference is not that big (identical maps taken out):
```
          compute-ro         compute.readonly
          compute-rw         compute
          computeaccounts-ro computeaccounts.readonly
          computeaccounts-rw computeaccounts
          logging-write      logging.write
          sql                sqlservice
          sql-admin          sqlservice.admin
          storage-full       devstorage.full_control
          storage-ro         devstorage.read_only
          storage-rw         devstorage.read_write
```
2) If it does make sense, maybe I should try to address this in Fog-google, so all fog-google based projects benefit?
3) If this should not be in fog, I'll address it in a separate issue as I was thinking of handling Fog config validation more gracefully anyways (like, not allowing for wrong data types to be entered and intercepting incorrect config before the disks are created), I think this'll fit well into that.

**Comment by erjohnso** [comment]
Putting it in fog-google makes the most sense, agreed.

As far as needing to keep a hand-maintained alias list up to date, you are correct. It would require that we keep fog-google updated if/when those mappings change.  I agree it's not ideal, so we just need to balance the tradeoff of having consistency with gcloud vs upkeep.  Keeping a list updated in fog-google that can be dynamically used in vagrant would be the best option imho.  That's actually the approach I've taken with libcloud and tools built on it.

(Oh, and hashing this out on GitHub is the right thing to do. Others can chime in with their thoughts that way.) ;-)

**Comment by Temikus** [comment]
Ack, let me open an issue in fog-google and see what the maintainers think about this.

**Pull request merged**